hack/test-cmd.sh

```diff
@@ -246,112 +246,16 @@ os::test::junit::declare_suite_start "cmd/config"
-VERBOSE=true os::cmd::expect_success 'oc get projects'
-VERBOSE=true os::cmd::expect_success 'oc project project-foo'
-os::cmd::expect_success_and_text 'oc config view' "current-context.+project-foo/${API_HOST}:${API_PORT}/test-user"
-os::cmd::expect_success_and_text 'oc whoami' 'test-user'
-os::cmd::expect_success_and_text "oc whoami --config='${MASTER_CONFIG_DIR}/admin.kubeconfig'" 'system:admin'
-os::cmd::expect_success_and_text 'oc whoami -t' '.'
-os::cmd::expect_success_and_text 'oc whoami -c' '.'
-
-# test config files from the --config flag
-os::cmd::expect_success "oc get services --config='${MASTER_CONFIG_DIR}/admin.kubeconfig'"
-
-# test config files from env vars
-os::cmd::expect_success "KUBECONFIG='${MASTER_CONFIG_DIR}/admin.kubeconfig' oc get services"
-
-# test completion command help
-os::cmd::expect_success_and_text "oc completion -h" "prints shell code"
-os::cmd::expect_success_and_text "openshift completion -h" "prints shell code"
-os::cmd::expect_success_and_text "oadm completion -h" "prints shell code"
-# test completion command output
-os::cmd::expect_failure_and_text "oc completion" "Shell not specified."
-os::cmd::expect_success "oc completion bash"
-os::cmd::expect_success "oc completion zsh"
-os::cmd::expect_failure_and_text "oc completion test_shell" 'Unsupported shell type "test_shell"'
-# test completion command for openshift
-os::cmd::expect_failure_and_text "openshift completion" "Shell not specified."
-os::cmd::expect_success "openshift completion bash"
-os::cmd::expect_success "openshift completion zsh"
-os::cmd::expect_failure_and_text "openshift completion test_shell" 'Unsupported shell type "test_shell"'
-# test completion command for oadm
-os::cmd::expect_failure_and_text "oadm completion" "Shell not specified."
-os::cmd::expect_success "oadm completion bash"
-os::cmd::expect_success "oadm completion zsh"
-os::cmd::expect_failure_and_text "oadm completion test_shell" 'Unsupported shell type "test_shell"'
-echo "oc completion: ok"
-
-# test config files in the home directory
-mkdir -p ${HOME}/.kube
-cp ${MASTER_CONFIG_DIR}/admin.kubeconfig ${HOME}/.kube/config
-os::cmd::expect_success 'oc get services'
-mv ${HOME}/.kube/config ${HOME}/.kube/non-default-config
-echo "config files: ok"
 os::test::junit::declare_suite_end
 
 # from this point every command will use config from the KUBECONFIG env var
+export KUBERNETES_MASTER="${API_SCHEME}://${API_HOST}:${API_PORT}"
 export NODECONFIG="${NODE_CONFIG_DIR}/node-config.yaml"
+mkdir -p ${HOME}/.kube
+cp ${MASTER_CONFIG_DIR}/admin.kubeconfig ${HOME}/.kube/non-default-config
 export KUBECONFIG="${HOME}/.kube/non-default-config"
 export CLUSTER_ADMIN_CONTEXT=$(oc config view --flatten -o template --template='{{index . "current-context"}}')
+oc login --server="${KUBERNETES_MASTER}" --certificate-authority="${MASTER_CONFIG_DIR}/ca.crt" -u test-user -p anything
```

Comment:
> By the time we run this script, we do not have a token: #10047 (comment)
> 
> I'm not entirely sure if it matters - if we were getting context from the login, we would fail the tests, as they expect client calls to fail until a successful login happens. I'm able to run hack/test-cmd.sh in its entirety locally without issues, so the other tests are correctly logged in as well. @liggitt do you have any insight on this? Have I invalidated some of the tests in this suite by not being able to logout at the start of it?

If you goal is to avoid side effects, it seems like you should be copying the kubeconfig and re-setting the `KUBECONFIG` envvar to avoid them.

Also, this is a token user.